### PR TITLE
fix(taskframework): running task be canceled incorrect due to heartbeat timeout 

### PR DESCRIPTION
**Pull request by krihy**
#### What type of PR is this?

type-bug
#### What this PR does / why we need it:
running task will be canceled due to heartbeat timeout,  executor will report `CANCELED` status to odc server, but task will be retry and running again, so odc server receive a incorrect signal and terminal task, so we ignore `CANCELED` result but receive `FAILED` status

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/Task.java [hunk 1]
```diff
@@ -40,6 +41,13 @@ public interface Task<RESULT> {
      */
     boolean stop();
 
+
+    /**
+     * Stop current task and set status to failed. This method will be called by {@link TaskMonitor} for
+     * stop a timeout task
+     */
+    boolean stopAndFailed();
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/Task.java:49]: how about `abort` or `timeout`.

Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/Task.java]: okey, rename to abort

File: server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java [hunk 2]
```diff
@@ -71,12 +71,21 @@ public void start(JobContext context) {
...
     public boolean stop() {
+        return stop(JobStatus.CANCELED);
+    }
+
+    @Override
+    public boolean stopAndFailed() {
+        return stop(JobStatus.FAILED);
+    }
+
+    private boolean stop(JobStatus status) {
```
[the remark this answers is not in the document]
Comment (krihy) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/task/executor/task/BaseTask.java]: add `synchronized` on this method

**Review by yizhouxw — APPROVED** [review]
lgtm

**Pull request merged**